packages/expo-modules-core/android/src/test/java/expo/modules/kotlin/KotlinInteropModuleRegistryTest.kt

```diff
@@ -19,65 +39,145 @@ class DummyModule_1 : Module() {
   override fun definition() = ModuleDefinition {
-    name("dummy-2")
+    name("test-2")
+    function("f1") {
+      throw TestException()
+    }
+    function("f2") { arg1: Int ->
+      arg1
+    }
     viewManager {
       view { mockk() }
     }
   }
 }
 
-val provider = object : ModulesProvider {
+private val provider = object : ModulesProvider {
   override fun getModulesList(): List<Class<out Module>> {
     return listOf(
-      DummyModule_1::class.java,
-      DummyModule_2::class.java
+      TestModule_1::class.java,
+      TestModule_2::class.java
     )
   }
 }
 
 class KotlinInteropModuleRegistryTest {
+  private val interopModuleRegistry = KotlinInteropModuleRegistry(
+    provider,
+    mockk(),
+    WeakReference(mockk(relaxed = true))
+  )
+
   @Test
   fun `should register modules from provider`() {
-    val interopModuleRegistry = KotlinInteropModuleRegistry(
-      provider,
-      mockk(),
-      WeakReference(mockk(relaxed = true))
-    )
-
-    interopModuleRegistry.hasModule("dummy-1")
-    interopModuleRegistry.hasModule("dummy-2")
+    interopModuleRegistry.hasModule("test-1")
+    interopModuleRegistry.hasModule("test-2")
   }
 
   @Test
   fun `should export constants`() {
-    val interopModuleRegistry = KotlinInteropModuleRegistry(
-      provider,
-      mockk(),
-      WeakReference(mockk(relaxed = true))
-    )
-
     Truth.assertThat(interopModuleRegistry.exportedModulesConstants())
       .containsExactly(
-        "dummy-1", mapOf("c1" to 123, "c2" to "123"),
-        "dummy-2", emptyMap<String, Any>()
+        "test-1", mapOf("c1" to 123, "c2" to "123"),
+        "test-2", emptyMap<String, Any>()
       )
   }
 
   @Test
   fun `should export view manages`() {
-    val interopModuleRegistry = KotlinInteropModuleRegistry(
-      provider,
-      mockk(),
-      WeakReference(mockk(relaxed = true))
-    )
-
     val rnManagers = interopModuleRegistry.exportViewManagers()
     val expoManagersNames = interopModuleRegistry.exportedViewManagersNames()
 
     Truth.assertThat(rnManagers).hasSize(1)
-    Truth.assertThat(rnManagers.first().name).isEqualTo("ViewManagerAdapter_dummy-2")
-    Truth.assertThat(expoManagersNames).containsExactly("dummy-2")
+    Truth.assertThat(rnManagers.first().name).isEqualTo("ViewManagerAdapter_test-2")
+    Truth.assertThat(expoManagersNames).containsExactly("test-2")
+  }
+
+  @Test
+  fun `call method should reject if something goes wrong`() {
+    val mockedPromise = PromiseMock()
+    val mockedPromise2 = PromiseMock()
+
+    interopModuleRegistry.callMethod("test-1", "f1", JavaOnlyArray(), mockedPromise)
+    interopModuleRegistry.callMethod("test-2", "f1", JavaOnlyArray(), mockedPromise2)
+
+    Truth.assertThat(mockedPromise.state).isEqualTo(PromiseState.REJECTED)
+    Truth.assertThat(mockedPromise.rejectMessage).isEqualTo(
+      """
+      Call to function 'test-1.f1' has been rejected.
+      → Caused by: java.lang.NullPointerException
+      """.trimIndent()
+    )
```

Comment:
Actually in Swift I stopped testing error messages. Imho we don't have to care as much what the message is, especially that having such tests is an additional cost for us. We can keep it though, just giving my opinion 😄